js/services/engine.ts

```diff
@@ -57,7 +57,9 @@ function processObjectDataRequests(components, flowKey) {
 
                 if (component.fileDataRequest) {
 
-                    return fileDataRequest(component.id, component.fileDataRequest, flowKey, limit);
+                    if (component.componentType !== 'file-upload') {
```

Comment:
This introduces implicit knowledge of a higher level from a lower level (i.e. from `ui-core` to `ui-bootstrap`) which is probably not a good idea - lower level dependencies and abstractions shouldn't be aware of anything higher than themselves.

In this case, it might be better for the component itself to be in charge of executing the file data request, rather than the lower level, as it's now required that the call doesn't happen every time (which was the previous assumption of this code).

Reply:
Thinking about it, there might be a straightforward way to delegate this back to the component. Have a look at the the uploadFiles function being called by the component [here](https://github.com/manywho/ui-core/blob/develop/js/services/connection.ts#L128) note that we are appending "FileDataRequest" to the form data being sent. I would check what happens if you remove this to see if a FileDataRequest key is returned in the response from the engine. If not then the component can specify whether or not to do this. Worth looking at.